lesson_04/main.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+type Shape interface {
+	Area() float64
+	Perimeter() float64
+}
+
+type Circle struct {
+	radius float64
+}
+
+func (c Circle) Area() float64 {
+	return math.Pow(c.radius, 2) * math.Pi
+}
+
+func (c Circle) Perimeter() float64 {
+	return 2 * c.radius * math.Pi
+}
+
+func (c Circle) String() string {
+	return fmt.Sprintf("Circle: radius %.2f", c.radius)
+}
+
+type Rectangle struct {
+	height float64
+	width  float64
+}
+
+func (r Rectangle) Area() float64 {
+	return r.height * r.width
+}
+
+func (r Rectangle) Perimeter() float64 {
+	return 2 * (r.width + r.height)
+}
+
+func (r Rectangle) String() string {
+	return fmt.Sprintf("\nRectangle with height %.2f and width %.2f", r.height, r.width)
+}
+
+func DescribeShape(s Shape) {
+	fmt.Println(s)
+	fmt.Printf("Area: %.2f\n", s.Area())
+	fmt.Printf("Perimeter: %.2f\n", s.Perimeter())
+}
+
+func main() {
+	// choose your own dimensions
+	c := Circle{radius: 8}
+	r := Rectangle{
+		height: 9,
+		width:  3,
+	}
+	DescribeShape(c)
+	DescribeShape(r)
+}
```

Comment:
Why is everything in one file?It was just an example. Please, move interface and entities in different files/packages